### PR TITLE
[XPU][SDPA] Add XPU flash_attention_forward dispatch with dropout fallback

### PR DESCRIPTION
**Pull request by gplutop7**
Fixes:
https://github.com/intel/torch-xpu-ops/issues/3698

This PR enables XPU support for aten._flash_attention_forward and adds a safe fallback path for dropout > 0.

- Adds XPU dispatch for aten._flash_attention_forward in native_functions.yaml.
- Implements _flash_attention_forward_xpu in the XPU attention backend.
- Keeps the existing fast SYCL-TLA path for dropout = 0.
- Adds a math-based fallback for dropout > 0 (including causal masking) to avoid unsupported kernel behavior.

python -m pytest test_meta_xpu.py::TestMetaXPU::test_meta_outplace_torch_ops_aten__flash_attention_forward_xpu_float16
PASSED

### CODE REVIEW

**Comment by guangyey** [comment]
cc @chunhuanMeng @LuFinch

**Review by guangyey — COMMENTED** [review]
Hi @gplutop7 
I understood, previously we fallback `_flash_attention_forward` to CPU, however, CPU acutally doesn't this op as well.

**Comment by LuFinch** [comment]
I have a PR to enable _flash_attention_forwrd/_backward https://github.com/pytorch/pytorch/pull/181559. And I am working on dropout SYCLTLA kernel. Please don't merge this PR.

**Comment by LuFinch** [comment]
Only add fallback for forward is unreasonable. Dropout is using for training and we should enable backward too.
